### PR TITLE
Update conformer with bug fixes.

### PR DESCRIPTION
**Pull request by VahidooX**

### CODE REVIEW

**Review by titu1994 — CHANGES_REQUESTED** [review]
Could you revert the changes related to WER for the time being? That needs further discussion and is not necessary for bug fixes to Conformer model itself. Everything else related to conformer modules looks good to me.

**Comment by VahidooX** [comment]
> 
> 
> Could you revert the changes related to WER for the time being? That needs further discussion and is not necessary for bug fixes to Conformer model itself. Everything else related to conformer modules looks good to me.

Dropped that part but keep in mind that predictions_len comes from the output of the decoder. So if a model return that, it is used otherwise it is ignored. For some models like encoder/decoder with cross attention it should return None for instance. So it would just used for the models who returns the valid length of their outputs.

**Comment by titu1994** [comment]
> Dropped that part but keep in mind that predictions_len comes from the output of the decoder. So if a model return that, it is used otherwise it is ignored. For some models like encoder/decoder with cross attention it should return None for instance. So it would just used for the models who returns the valid length of their outputs.

All CTC based encoders return this length value, which would affect WER calculations wrt QuartzNet/Jasper as well. In the future, yes we can add this but there needs to be a special flag passed by only certain models - say Attention based models - which then submits the lengths for masking to the WER decoding. This is so that unrelated models such as QN/Jasper are not affected by this masking during wer.

File: nemo/collections/asr/models/ctc_models.py [hunk 1]
```diff
@@ -393,7 +393,7 @@ def forward(
     def training_step(self, batch, batch_nb):
         signal, signal_len, transcript, transcript_len = batch
         if isinstance(batch, DALIOutputs) and batch.has_processed_signal:
-            log_probs, encoded_len, predictions = self.forward(
+            log_probs, predictions_len, predictions = self.forward(
```
Comment (titu1994) [inline nemo/collections/asr/models/ctc_models.py:396]: This needs to remain encoded_len

**Review by titu1994 — COMMENTED** [review]
One last change, then good to go.

File: nemo/collections/asr/models/ctc_models.py (hunk 1 above)
Comment (VahidooX) [reply nemo/collections/asr/models/ctc_models.py]: Updated.

**Review by titu1994 — APPROVED** [review]
LGTM

**Pull request merged**